test/idp/src/main/kotlin/org/codice/compliance/web/sso/RedirectLoginTest.kt

```diff
@@ -94,11 +98,15 @@ class RedirectLoginTest : StringSpec() {
                     .get(Common.getSingleSignOnLocation(SamlProtocol.REDIRECT_BINDING))
 
             // Get response from plugin portion
-            val idpResponse = getServiceProvider(IdpResponder::class.java).getIdpRedirectResponse(response).apply {
-                isRelayStateGiven = true
-            }
+            val idpResponse = getServiceProvider(IdpResponder::class.java)
+                    .getIdpRedirectResponse(response).decorate().apply {
+                        isRelayStateGiven = true
+                    }
+
+            val bindingVerifier = idpResponse.bindingVerifier()
+            bindingVerifier.verify()
 
-            val responseDom = verifyResponse(idpResponse)
+            val responseDom = idpResponse.responseDom!!
```

Comment:
❗️ This can potentially through an NPE (protection against which is, please recall, one of the primary advantages that Kotlin provides over Java).

See note on `IdpResponseDecorators.kt`.